### PR TITLE
Add script to migrate features to BCD tags

### PR DESCRIPTION
**Pull request by queengooborg**
This PR is a continuation off of #473, which adds a script that migrates the data for the BCD features that are included in each feature to tags within BCD.  To run, simply run `node --loader=ts-node/esm scripts/migrate-to-bcd.ts ../browser-compat-data` or `npx tsx scripts/migrate-to-bcd.ts ../browser-compat-data`.

The key difference between this PR and the original is that the BCD path has been replaced to use a relative path by default, which expects the BCD folder to be right next to the `web-features` folder -- but more importantly, the path can be overridden with the `BCD_PATH` environment variable.

Closes #473.

### CODE REVIEW

**Review by Elchi3 — COMMENTED** [review]
Thanks Vinyl! The BCD path works great for me.

I think we want to rename the file from migrate.ts to migrate-to-bcd.ts.

The script doesn't handle renames or removals. How hard would it be to implement that? See https://github.com/web-platform-dx/web-features/pull/1605 that renamed growable-sharedarraybuffer to resizable-buffers, for example. If I run migrate, I end up with 
```json
"tags": [
  "web-features:snapshot:ecmascript-2024",
  "web-features:growable-sharedarraybuffer",
  "web-features:resizable-buffers"
],
```
Desired would be:

```json
"tags": [
  "web-features:snapshot:ecmascript-2024",
  "web-features:resizable-buffers"
],
```

**Comment by Elchi3** [comment]
Thanks, that works great for me!

The tags are added to the bottom of the JSON in BCD. Should we run "npm run fix" in BCD in the script somehow to fix that?

**Comment by queengooborg** [comment]
We could do that, but I think that it'd be better to do so outside of the script -- for example, during a GitHub Actions workflow.  What do you think?

**Review by Elchi3 — APPROVED** [review]
Either way would be fine with me. 
Approving but waiting for another owner to review this PR before merging.

File: scripts/migrate-to-bcd.ts
Comment (ddbeck) [inline scripts/migrate-to-bcd.ts:23]: This is now enforced in [`index.ts`#L170-L180](https://github.com/web-platform-dx/web-features/blob/9eb029a6adc104588d4b9c45819722f064e5075b/index.ts#L170-L180), so we don't need to duplicate this check here.

```suggestion
```

File: scripts/migrate-to-bcd.ts
Comment (ddbeck) [inline scripts/migrate-to-bcd.ts:39]: Style nit: we've been mostly using `function` in scripts here. Would you mind re-doing this as a plain function?

File: scripts/migrate-to-bcd.ts
Comment (ddbeck) [inline scripts/migrate-to-bcd.ts:89]: Use a `winston` logger here (see my earlier comment about the CLI setup), so we get the nice formatting.

File: scripts/migrate-to-bcd.ts
Comment (queengooborg) [reply scripts/migrate-to-bcd.ts]: The mdn-bcd-collector uses an environment variable, so I thought it would be best to use an environment variable for consistency with other tooling.  If you'd really like me to use `yargs`, I can, although since the goal is to run this script automatically using GitHub Actions, I don't think it will make much of a difference.

Comment (ddbeck) [reply scripts/migrate-to-bcd.ts]: Yes, I would. It's mostly about about help/documentation and the principle of least surprise. Env vars as the sole method of runtime configuration is unusual for web-features, even if it's not for other projects. This isn't true for everything, but I'd expect something like `npm run migrate-to-bcd -- --help` or `npm run migrate-to-bcd -- --bcd-dir=…` to do something useful.

(I guess now's a good time to add this: I think having a default directory is quite surprising too. It didn't _work_ but it also didn't _fail_ the first time I ran it because I don't have BCD checked out at the default path. I think requiring a path would be much less surprising and doing so with yargs would get us help and error messages too.)

Also, using env vars is still possible, so it doesn't conflict with the collector's conventions. You can set yargs options with env vars, if you set a prefix (which is also a good practice): https://yargs.js.org/docs/#api-reference-envprefix. That would be quite friendly to GitHub Actions and tools like `direnv`.

Comment (ddbeck) [reply scripts/migrate-to-bcd.ts]: @queengooborg do you plan to come back to this?

**Comment by Elchi3** [comment]
@queengooborg can you resolve the conflicts here?

**Review by ddbeck — APPROVED** [review]

**Pull request merged**
